### PR TITLE
[docs] add status header for redaction

### PR DESCRIPTION
**Pull request by Frapschen**
**Description:** add status header for redaction

### CODE REVIEW

**Comment by djaglowski** [comment]
Code owners @leonsp-ai, @dmitryax, @mx-psi, please verify the proposed status.

**Review by mx-psi — COMMENTED** [review]
This component is not available on any distribution, so I think it should be marked as 'in development'

**Comment by dmitryax** [comment]
> This component is not available on any distribution, so I think it should be marked as 'in development'

Makes sense to me. 

@leonsp-ai do you want to add the component in [the contrib distribution](https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/distributions/otelcol-contrib/manifest.yaml)?

**Comment by dmitryax** [comment]
Added the processor to the build https://github.com/open-telemetry/opentelemetry-collector-releases/pull/185

File: processor/redactionprocessor/README.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Redaction processor
 
-Supported pipeline types: traces
+| Status                   |           |
+| ------------------------ |-----------|
+| Stability                | [beta]    |
```
Comment (dmitryax) [inline processor/redactionprocessor/README.md:5]: Let's call it alpha for now

**Review by dmitryax — APPROVED** [review]

**Pull request merged**